com.unity.render-pipelines.high-definition/Runtime/Material/SharedRTManager.cs

```diff
@@ -439,38 +442,56 @@ public void BindNormalBuffer(CommandBuffer cmd, bool isMSAA = false)
             cmd.SetGlobalTexture(HDShaderIDs._NormalBufferTexture, GetNormalBuffer(isMSAA));
         }
 
+        private bool CompareRTHandle(RTHandle a, RTHandle b)
+        {
+            return (a != null && b != null && a.rt.graphicsFormat == b.rt.graphicsFormat);
+        }
+
         public void ResolveSharedRT(CommandBuffer cmd, HDCamera hdCamera)
         {
             if (hdCamera.frameSettings.IsEnabled(FrameSettingsField.MSAA))
             {
                 Debug.Assert(m_MSAASupported);
                 using (new ProfilingScope(cmd, ProfilingSampler.Get(HDProfileId.ResolveMSAADepth)))
                 {
-                    if (m_MotionVectorsSupport)
+                    if (DeviceInfo.requiresExplicitMSAAResolve == false &&
+                        (m_MotionVectorsSupport == true ? CompareRTHandle(m_MotionVectorsMSAART, m_MotionVectorsRT) : true) &&
+                        CompareRTHandle(m_NormalMSAART, m_NormalRT) &&
+                        CompareRTHandle(m_DepthAsColorMSAART, m_CameraDepthValuesBuffer))
                     {
-                        // Grab the RTIs and set the output render targets
-                        m_RTIDs3[0] = m_CameraDepthValuesBuffer.nameID;
-                        m_RTIDs3[1] = m_NormalRT.nameID;
-                        m_RTIDs3[2] = m_MotionVectorsRT.nameID;
-                        CoreUtils.SetRenderTarget(cmd, m_RTIDs3, m_CameraDepthStencilBuffer);
-
-                        // Set the motion vector input texture
-                        Shader.SetGlobalTexture(HDShaderIDs._MotionVectorTextureMS, m_MotionVectorsMSAART);
+                        if (m_MotionVectorsSupport)
+                            cmd.ResolveAntiAliasedSurface(m_MotionVectorsMSAART.rt, m_MotionVectorsRT.rt);
```

Comment:
I don't see how this code can work. We can't rely on "auto" resolve for our buffer. If we use a shader for the resolve, it is because we need to do particular software operation to do the resolve (like taking the closted depth sample), we don't blindly average vector value for example.

See 
D:\newSRP\Graphics\com.unity.render-pipelines.high-definition\Runtime\RenderPipeline\RenderPass\MSAA\DepthValues.shader

So it mean we are block for Metal? auto resolve will give us poor result and we can't used our software resolve? or it was just an optimization?